### PR TITLE
Create wrapper for Material-UI Autocomplete

### PR DESCRIPTION
**Pull request by abrezuica**
This is a new wrapper we created for the _Material-UI_ Autocomplete component:
https://material-ui.com/api/autocomplete

In the previous Autocomplete, we were using _react-select_. This version was using some old implementations and was hard to debug, so we decided to switch to the Material-UI one, even though it is still in lab.

For the moment, we have managed to cover all the old functionalities that the old component had and come with some fixes and new features as well.

### **Important differences**

**Problems solved:**
- the lazy Autocomplete with a value already selected (for example, in an edit screen) displays the value in the input
- easier to debug, the old Autocomplete was made up of more components; now we are using only one Autocomplete, with different states and behaviors
- the _creatable_ functionality was available only for the lazy-loading Autocomplete; now it is working in all the scenarios

**New functionalities:**
- can receive an array of string values as options (previously, it could accept only an array of objects)
- we can use home/end keys to move focus to the first/last item in the options list
- uses the Material-UI Chip component for showing the selected values when the Autocomplete is multi-selection

**Deprecated props:**

- _isSearchable_ - this prop is deprecated, as it doesn't need to be explicitly sent to the component; all the autocompletes are searchable by default.

**New props:**

- _withCheckboxes_ - used when the Autocomplete is multi-selection: we can choose if we want to have checkboxes in the list of options or not

### CODE REVIEW

File: __tests__/Autocomplete/__snapshots__/Autocomplete.test.js.snap [hunk 1]
```diff
@@ -2,85 +2,122 @@
...
+  className="MuiAutocomplete-root MuiAutocomplete-fullWidth MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
+  label="Status"
+  onClick={[Function]}
   onKeyDown={[Function]}
+  onMouseDown={[Function]}
+  role="combobox"
 >
   <div
-    className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
+    className="MuiFormControl-root MuiTextField-root undefined undefined MuiFormControl-fullWidth"
```
Comment (andra-sava) [inline __tests__/Autocomplete/__snapshots__/Autocomplete.test.js.snap:15]: duplicated "undefined"

File: __tests__/Autocomplete/Autocomplete.test.js [hunk 2]
```diff
@@ -1,24 +1,177 @@
...
+      <Autocomplete
+        label={"Status"}
+        simpleValue
+        onChange={jest.fn()}
+        loadOptions={loadOptions}
+        defaultOptions
+      />
+    );
+
+    expect(loadOptions).toBeCalled;
```
Comment (abrezuica) [inline __tests__/Autocomplete/Autocomplete.test.js:102]: toBeCalled()

File: __tests__/Autocomplete/Autocomplete.test.js [hunk 3]
```diff
@@ -1,24 +1,177 @@
...
+    mount(
+      <Autocomplete
+        label={"Status"}
+        simpleValue
+        onChange={jest.fn()}
+        loadOptions={loadOptions}
+      />
+    );
+
+    expect(loadOptions.mock.calls.length).toBe(0);
```
Comment (abrezuica) [inline __tests__/Autocomplete/Autocomplete.test.js:119]: maybe .not.toBeCalled()

File: __tests__/Autocomplete/Autocomplete.test.js [hunk 4]
```diff
@@ -1,24 +1,177 @@
...
+        isClearable
+        value={"2"}
+        onChange={onChange}
+      />
+    );
+
+    const muiAutocomplete = wrapper.find(MuiAutocomplete);
+    muiAutocomplete.props().onChange({ target: { value: "1" } });
+
+    expect(onChange).toBeCalled;
```
Comment (abrezuica) [inline __tests__/Autocomplete/Autocomplete.test.js:175]: wrong test

File: components/Autocomplete/Autocomplete.d.ts [hunk 5]
```diff
@@ -1,90 +1,94 @@
...
    */
-  options?: []
+  options: [OptionType | string]
   /**
-  * Function that returns a promise, which is the set of options to be used once the promise resolves.
-  */
-  loadOptions(inputValue: string, callback: (options: OptionsType<OptionType>) => void): void;
+   * Function that returns a promise, which is the set of options to be used once the promise resolves.
+   */
+  loadOptions?: (() => Promise)
```
Comment (abrezuica) [inline components/Autocomplete/Autocomplete.d.ts:19]: Promise<[OptionType | string]>

**Review by dragos-rosca — APPROVED** [review]

**Pull request merged**